py-polars/polars/internals/dataframe/frame.py

```diff
@@ -3811,19 +3780,21 @@ def head(self, n: int = 5) -> Self:
         │ 1   ┆ 6   ┆ a   │
         │ 2   ┆ 7   ┆ b   │
         └─────┴─────┴─────┘
+
         """
         if n < 0:
-            n = len(self) + n
+            n = max(0, self.height + n)
```

Comment:
This is the fix - the rest is just cleanup :)

Reply:
👍